utils/handles/signal.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __UTILS_HANDLES_SIG_H__
+#define __UTILS_HANDLES_SIG_H__
+
+/**
+ * We declare a function register_signal() to handle signals while the program
+ * executing. This function will catch some exceptional behavior with the
+ * program.
+ */
+
+#include <signal.h>
+#include <stdlib.h>
+
+typedef enum UNIVERSAL_SINGAL_NUM { null, ctrl_c } universal_singal_num_t;
+
+#if !defined(_WIN32) && defined(__unix__) || defined(__unix) || \
+    (defined(__APPLE__) && defined(__MACH__))
+
+#include <unistd.h>
+#define ctrl_c_signal SIGINT
+#define signal_error SIG_ERR
+
+void signal_handler_posix(int signo);
+void (*end_core)(universal_singal_num_t);
+
+static inline __sighandler_t register_signal(
+    int SIG, void (*handler_core_end)(universal_singal_num_t)) {
+  end_core = handler_core_end;
```

Comment:
Names shouldn't be related to core, they should be generic

Reply:
Hi since I thought the std <signal.h> passes a function as signal handler, therefore, I imitated the std <signal.h> pass a function.
should I mvoe the signal handler in `main.c` to `utils/handles/signal.c` ?